virtual_ecosystem/models/abiotic/wind.py

```diff
@@ -128,685 +116,88 @@ def calculate_roughness_length_momentum(
-
-    Args:
-        canopy_height: Canopy height, [m]
-        leaf_area_index: Leaf area index, [m m-1]
-        mean_mixing_length: Mixing length for canopy air transport, [m]
-        drag_coefficient: Drag coefficient, dimensionless
-        relative_turbulence_intensity: Relative turbulence intensity, dimensionless
-
-    Returns:
-        Wind attenuation coefficient, dimensionless
-    """
-
-    # VIVI - this is operating on inputs containing all true aboveground rows. Because
-    # LAI is only defined for the canopy layers, the result of this operation is
-    # undefined for the top and bottom row and so can just be filled in rather than
-    # having to concatenate. We _could_ subset the inputs and then concatenate - those
-    # are more intuitive inputs - but handling those extra layers maintains the same
-    # calculation shape throughout the wind calculation stack.
-    attenuation_coefficient = (drag_coefficient * leaf_area_index * canopy_height) / (
-        2 * mean_mixing_length * relative_turbulence_intensity
-    )
-
-    # Above the canopy is set to zero and the surface layer is set to the last valid
-    # canopy value
-    attenuation_coefficient[0] = 0
-    attenuation_coefficient[-1] = find_last_valid_row(attenuation_coefficient)
-
-    return attenuation_coefficient
-
+    # Ensure that heights are greater than roughness length and zero_plane_displacement
+    # to avoid division by zero or negative logarithm
+    heights = np.maximum(wind_heights, roughness_length + 1e-5)
+    heights = np.maximum(wind_heights, zero_plane_displacement + 1e-5)
 
-def wind_log_profile(
-    height: float | NDArray[np.float32],
-    zeroplane_displacement: float | NDArray[np.float32],
-    roughness_length_momentum: float | NDArray[np.float32],
-    diabatic_correction_momentum: float | NDArray[np.float32],
-) -> NDArray[np.float32]:
-    """Calculate logarithmic wind profile.
-
-    Note that this function can return NaN, this is not corrected here because it might
-    cause division by zero later on in the work flow.
-
-    Args:
-        height: Array of heights for which wind speed is calculated, [m]
-        zeroplane_displacement: Height above ground within the canopy where the wind
-            profile extrapolates to zero, [m]
-        roughness_length_momentum: Momentum roughness length, [m]
-        diabatic_correction_momentum: Diabatic correction factor for momentum
-
-    Returns:
-        logarithmic wind profile
-    """
-
-    wind_profile = (
-        np.log((height - zeroplane_displacement) / roughness_length_momentum)
-        + diabatic_correction_momentum,
+    wind_speed = (
+        reference_wind_speed
+        * np.log((heights - zero_plane_displacement) / roughness_length)
+        / np.log((reference_height - zero_plane_displacement) / roughness_length)
     )
-
-    return np.where(wind_profile == 0.0, np.nan, wind_profile).squeeze()
+    return np.where(wind_speed >= 0, wind_speed, min_wind_speed)
```

Comment:
Shouldn't this be something like:

```suggestion
    return np.where(wind_speed > min_wind_speed, wind_speed, min_wind_speed)
```

Otherwise, you could get values below the minimum wind speed, but that are just greater than zero?